parser/parser.go

```diff
@@ -41,24 +41,25 @@ type parser struct {
 	cfg *Config
 
 	// accumulated results
-	fset          *token.FileSet
-	errors        *errlist.List
-	pkgs          []*est.Package
-	pkgMap        map[string]*est.Package // import path -> pkg
-	svcs          []*est.Service
-	jobs          []*est.CronJob
-	svcMap        map[string]*est.Service // name -> svc
-	svcPkgPaths   map[string]*est.Service // pkg path -> svc
-	jobsMap       map[string]*est.CronJob // ID -> job
-	pubSubTopics  []*est.PubSubTopic
-	cacheClusters []*est.CacheCluster
-	names         names.Application
-	authHandler   *est.AuthHandler
-	middleware    []*est.Middleware
-	declMap       map[string]*schema.Decl // pkg/path.Name -> decl
-	decls         []*schema.Decl
-	paths         paths.Set                          // RPC paths
-	resourceMap   map[string]map[string]est.Resource // pkg/path -> name -> resource
+	fset                        *token.FileSet
+	errors                      *errlist.List
+	pkgs                        []*est.Package
+	pkgMap                      map[string]*est.Package // import path -> pkg
+	svcs                        []*est.Service
+	jobs                        []*est.CronJob
+	svcMap                      map[string]*est.Service // name -> svc
+	svcPkgPaths                 map[string]*est.Service // pkg path -> svc
+	jobsMap                     map[string]*est.CronJob // ID -> job
+	pubSubTopics                []*est.PubSubTopic
+	cacheClusters               []*est.CacheCluster
+	names                       names.Application
+	authHandler                 *est.AuthHandler
+	middleware                  []*est.Middleware
+	declMap                     map[string]*schema.Decl // pkg/path.Name -> decl
+	decls                       []*schema.Decl
+	paths                       paths.Set                          // RPC paths
+	resourceMap                 map[string]map[string]est.Resource // pkg/path -> name -> resource
+	structsWithUnexportedFields map[*schema.Struct]*ast.Field      // A struct will be in this map if it has unexported fields
```

Comment:
This is really long compared with all the other fields which makes it quite awkward. How about:
```suggestion
	hasUnexportedFields map[*schema.Struct]*ast.Field      // A struct will be in this map if it has unexported fields
```